src/test/java/org/prebid/server/bidder/adnuntius/AdnuntiusBidderTest.java

```diff
@@ -0,0 +1,410 @@
+
+        // then
+        assertThat(result.getValue()).hasSize(0);
+        assertThat(result.getErrors()).hasSize(1)
+                .extracting(BidderError::getMessage)
+                .allMatch(errorMessage -> errorMessage.startsWith("Unmarshalling error:"));
+    }
+
+    @Test
+    public void makeHttpRequestsShouldReturnRequestsWithAdUnitsSeparatedByImpExtNetwork() {
+        // given
+        final BidRequest bidRequest = givenBidRequest(
+                givenImp(ExtImpAdnuntius.of("auId1", null), identity()),
+                givenImp(ExtImpAdnuntius.of("auId2", null), identity()),
+                givenImp(ExtImpAdnuntius.of("auId1", "network"), identity()),
+                givenImp(ExtImpAdnuntius.of("auId2", "network"), identity()));
+
+        // when
+        final Result<List<HttpRequest<AdnuntiusRequest>>> result = bidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getValue()).hasSize(2)
+                .extracting(HttpRequest::getPayload)
+                .extracting(AdnuntiusRequest::getAdUnits)
+                .allSatisfy(adUnits -> assertThat(adUnits).hasSize(2)
+                        .extracting(AdnuntiusAdUnit::getAuId)
+                        .containsExactly("auId1", "auId2"));
+        assertThat(result.getErrors()).hasSize(0);
+    }
+
+    @Test
+    public void makeHttpRequestsShouldReturnRequestsWithCorrectAdUnits() {
+        // given
+        final BidRequest bidRequest = givenBidRequest(givenImp(identity()),
+                givenImp(ExtImpAdnuntius.of("auId", null), identity()),
+                givenImp(ExtImpAdnuntius.of("auId", null), imp -> imp.id("impId")));
+
+        // when
+        final Result<List<HttpRequest<AdnuntiusRequest>>> result = bidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getValue()).hasSize(1)
+                .extracting(HttpRequest::getPayload)
+                .flatExtracting(AdnuntiusRequest::getAdUnits)
+                .extracting(AdnuntiusAdUnit::getAuId, AdnuntiusAdUnit::getTargetId)
+                .containsExactly(tuple(null, "null-null"), tuple("auId", "auId-null"), tuple("auId", "auId-impId"));
+        assertThat(result.getErrors()).hasSize(0);
+    }
+
+    @Test
+    public void makeHttpRequestsShouldReturnRequestsWithMetaDataIfUserIdIsPresent() {
+        // given
+        final BidRequest bidRequest = givenBidRequest(request -> request.user(User.builder().id("userId").build()),
+                givenImp(ExtImpAdnuntius.of(null, "network"), identity()), givenImp(identity()));
+
+        // when
+        final Result<List<HttpRequest<AdnuntiusRequest>>> result = bidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getValue()).hasSize(2)
+                .extracting(HttpRequest::getPayload)
+                .extracting(AdnuntiusRequest::getMetaData)
+                .extracting(AdnuntiusMetaData::getUsi)
+                .containsOnly("userId");
```

Comment:
We try not to use contains only, replace with call to `containsExactly`, it's more expicit. Also, please, check for similar occurences.